.azure/pipelines/jobs/project-ci.yml

```diff
@@ -13,37 +13,61 @@
 #       The matrix of configurations to run. By default, it runs a Debug and Release build on all platforms
 #   codeSign: boolean
 #       This build definition is enabled for code signing. (Only applies to Windows)
+#   runCodeQL3000: boolean
+#       This build should run CodeQL3000 instead of the regular build
 
 parameters:
   buildArgs: ''
   beforeBuild: []
   afterBuild: []
   codeSign: false
   variables: {}
+  runCodeQL3000: false
 
 jobs:
-- template: default-build.yml
-  parameters:
-    agentOs: Windows
-    matrix: ${{ parameters.matrix }}
-    buildArgs: ${{ parameters.buildArgs }}
-    beforeBuild: ${{ parameters.beforeBuild }}
-    afterBuild: ${{ parameters.afterBuild }}
-    codeSign: ${{ parameters.codeSign }}
-    variables: ${{ parameters.variables }}
-- template: default-build.yml
-  parameters:
-    agentOs: macOS
-    matrix: ${{ parameters.matrix }}
-    buildArgs: ${{ parameters.buildArgs }}
-    beforeBuild: ${{ parameters.beforeBuild }}
-    afterBuild: ${{ parameters.afterBuild }}
-    variables: ${{ parameters.variables }}
-- template: default-build.yml
-  parameters:
-    agentOs: Linux
-    matrix: ${{ parameters.matrix }}
-    buildArgs: ${{ parameters.buildArgs }}
-    beforeBuild: ${{ parameters.beforeBuild }}
-    afterBuild: ${{ parameters.afterBuild }}
-    variables: ${{ parameters.variables }}
+- ${{ if and(ne(variables['System.TeamProject'], 'public'), eq(parameters.runCodeQL3000, 'true')) }}:
+  - template: default-build.yml
+    parameters:
+      agentOs: Windows
+      matrix: ${{ parameters.matrix }}
+      buildArgs: ${{ parameters.buildArgs }}
+      beforeBuild:
```

Comment:
Reorder the `beforeBuild` steps and add the usual `condition` to the `script` step